service/src/main/java/bio/terra/catalog/service/SchemaService.java

```diff
@@ -0,0 +1,38 @@
+package bio.terra.catalog.service;
+
+import bio.terra.catalog.config.SchemaConfiguration;
+import bio.terra.common.exception.BadRequestException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.Set;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SchemaService {
```

Comment:
This might not be the best name.. this is a service that validates dataset metadata against a fixed schema. So it's not really a general schema service but one specific to dataset metadata. Although I guess in the future we might have other schemas that we validate with? So maybe `JsonValidationService`, since it what it does is validate json. The fact that it uses a schema to do it isn't necessarily important to the caller.